test/conformance/conformance_parallel_reduce.cpp

```diff
@@ -174,3 +206,42 @@ TEST_CASE("parallel_[deterministic_]reduce and std::invoke") {
 }
 
 #endif
+
+TEST_CASE("test rvalue optimization") {
+    constexpr std::size_t n_vectors = 10000;
+    std::vector<vector_wrapper<int>> vector_of_vectors;
+    auto init = {1, 2, 3, 4, 5};
+
+    vector_of_vectors.reserve(n_vectors);
+    for (std::size_t i = 0; i < n_vectors; ++i) {
+        vector_of_vectors.emplace_back(vector_wrapper<int>(init));
+    }
+
+    oneapi::tbb::blocked_range<std::size_t> range(0, n_vectors);
+
+    auto reduce_body = [&](const oneapi::tbb::blocked_range<std::size_t>& rng, vector_wrapper<int>&& x) {
+        vector_wrapper<int> new_vector = std::move(x);
+        for (std::size_t index = rng.begin(); index != rng.end(); ++index) {
+            new_vector.reserve(vector_of_vectors[index].size());
+            new_vector.insert(new_vector.end(), std::make_move_iterator(vector_of_vectors[index].begin()), std::make_move_iterator(vector_of_vectors[index].end()));
+        }
+        return new_vector;
+    };
+
+    auto join_body = [](vector_wrapper<int>&& x, vector_wrapper<int>&& y) {
+        vector_wrapper<int> new_vector = std::move(x);
+        new_vector.reserve(new_vector.size() + y.size());
+        new_vector.insert(new_vector.end(), std::make_move_iterator(y.begin()), std::make_move_iterator(y.end()));
```

Comment:
A move iterator over a vector of `int`s doesn't buy much. Consider for this example making
```cpp
struct MoveOnlyInt {
    int x;
    MoveOnlyInt() = default;
    explicit MoveOnlyInt(int x) : x{x} {}
    MoveOnlyInt(const MoveOnlyInt&) = delete;
    MoveOnlyInt(MoveOnlyInt&& other) : x{std::exchange(other.x, 0)} {}
    MoveOnlyInt& operator=(const MoveOnlyInt&) = delete;
    MoveOnlyInt& operator=(MoveOnlyInt&& other) {
        this->x = std::exchange(other.x, 0);
        return *this;
    }
};
```
and use vectors of `MoveOnlyInt` so the move iterators are required?

Reply:
Could also consider a test case using `std::list` or `std::set` that moves the nodes across, like
```cpp
auto join_body = [](std::set<int>&& x, std::set<int>&& y) {
   while (!y.empty()) {
       x.insert(y.extract(y.begin()));
   }
   return std::move(x);
}
```
(That could use `MoveOnlyInt` too.)

Reply:
It would probably be good to do a test with a type that isn’t default constructible and/or isn’t movable or copyable in a node-based container — basically all the weird corner cases of irregular types.

Reply:
Unfortunately, we still cannot use move-only types for parallel_reduce since we need to copy the identity into each reduction "leaf". And using `std::vector<MoveOnlyInt>` or list would cause compilation errors even if the identity element is empty. 
I have changed the value type into the integer wrapper that is copyable, but the actual copy constructor should never be called.

Regarding adding new test cases, I think it makes no sense to test both "vector of vectors" and "vector of lists" use-cases. Splicing the lists while doing parallel_reduce is more interesting IMHO, so I left only the vector of lists test.